sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -109,18 +128,18 @@ class BulkWriter(container: CosmosAsyncContainer,
 
           if (resp.getException != null) {
             Option(resp.getException) match {
-              case Some(cosmosException: CosmosException) => {
-                log.logDebug(s"encountered ${cosmosException.getStatusCode}")
+              case Some(cosmosException: CosmosException) =>
+                log.logDebug(s"encountered ${cosmosException.getStatusCode}, Context: ${operationContext.toString}")
```

Comment:
does toString get invoked here even if debug log is not enabled?

Reply:
Don't think so - at least my understanding of sclaa string interpolation is that it would generate a function at compile time that would only be invoked in this case if the simple logger actually logs the message?

Reply:
Actually not true = Scala string interpolation would be executed beforehand

Reply:
I think this is something we should change across the board - at least for logDebug?

Reply:
never mind - logDebug is doing lazy evaluation - all good